### ISSUE TITLE
x.detach() in GNNencoder

### ISSUE THREAD

**Issue by ZZhangsm** [EdisonLeeeee/lrGAE#1]
I noticed that in **encoders.py, at line 83**, you use `out.append(x.detach())`. 

 This implies that the GNNencoder is detached from the computation graph and, as a result, is not being updated during backpropagation.  Could you clarify if the encoder is supposed to be trainable?

**Comment by EdisonLeeeee** [comment]
Thank you for pointing that out. You are right, and this line of code was mistakenly added during my experimental probing. Will fix it now.
